music_service_async_interface/__init__.py

```diff
@@ -147,27 +198,58 @@ async def parse_urls(self, long_string: str) -> AsyncGenerator["Object", None]:
 
 
 class Object(ABC):
+    """Abstract class representing music service object eg. Track, Artist or Playlist
+    Should be subclassed on API implementation of music service.
+    """
+
     sess: Session
 
     @classmethod
     @abstractmethod
     async def from_url(cls, sess: Session, url: str) -> "Object":
-        """
-        :raises: `InvalidURL` -- when URL is being unparsable by this `Object`
+        """Fetches corresponding object from music service based on URL
+
+        :param sess: :class:`Session` instance to use when loading data from music service
+        :param url: music service URL pointing to object
+        :raises NotImplementedError: when particular object type can't be fetched by URL
+        :raises InvalidURL: when URL is being unparsable by this :class:`Object`
+        :return: corresponding object, eg. :class:`Track` or :class:`Playlist`
         """
         raise NotImplementedError
 
     @classmethod
     @abstractmethod
     async def from_id(cls, sess: Session, id_) -> "Object":
+        """Fetches corresponding object from music service based on ID
+
+        :param sess: :class:`Session` instance to use when loading data from music service
+        :param id_: music service ID of :class:`Object`
+        :return: corresponding :class:`Object`
+        """
         ...
 
     @abstractmethod
     async def get_url(self) -> str:
+        """Gets object's URL
+
+        :return: URL to object
+        """
         ...
 
     @abstractmethod
     def get_id(self):
+        """Gets object's music service ID
+
+        :return: music service ID of object
```

Comment:
what's with the music service id? What is it?

Reply:
same as above